apis/src/components/atoms/history_button.rs

```diff
@@ -73,4 +74,48 @@ fn send_action(action: &HistoryNavigation) {
             game_state_signal.view_game()
         }
     }
+    set_timer_from_response();
+}
+
+pub fn set_timer_from_response() {
+    let game_state_signal = expect_context::<GameStateSignal>();
+    let timer = expect_context::<TimerSignal>();
+    let is_finished = game_state_signal.is_finished();
+    //don't update timer if game isnt finished or for correspondence games
+    if !is_finished()
+        || matches!(
+            game_state_signal.signal.get().get_game_speed(),
```

Comment:
game_state_signal.signal.get().game_response.as_ref().is_some_and(|gr| gr.time_mode == TimeMode::RealTime) also works, to me that seems better instead of the match. I might be wrong tho.